common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -127,31 +193,16 @@ def list(self, request, *args, **kwargs):
 
     def perform_destroy(self, instance):
         """
-        This method is an override and is called by the DELETE method
+        This method is an override and is called by the destroy method, which is called when a DELETE operation occurs
+
+        This method will revoke the User's entitlement and unenroll the user if they are enrolled
+        in a Course Run
         """
-        save_model = False
-        if instance.expired_at is None:
-            instance.expired_at = timezone.now()
-            log.info('Set expired_at to [%s] for course entitlement [%s]', instance.expired_at, instance.uuid)
-            save_model = True
-
-        if instance.enrollment_course_run is not None:
-            CourseEnrollment.unenroll(
-                user=instance.user,
-                course_id=instance.enrollment_course_run.course_id,
-                skip_refund=True
-            )
-            enrollment = instance.enrollment_course_run
-            instance.enrollment_course_run = None
-            save_model = True
-            log.info(
-                'Unenrolled user [%s] from course run [%s] as part of revocation of course entitlement [%s]',
-                instance.user.username,
-                enrollment.course_id,
-                instance.uuid
-            )
-        if save_model:
-            instance.save()
+        log.info(
+            'Entitlement Revoke requested for Course Entitlement[%s]',
+            instance.uuid
+        )
+        _process_revoke_and_unenroll_entitlement(instance)
```

Comment:
Nit: It might be worth adding a comment explaining that this should not be called with the is_refund flag, as this endpoint is only meant to be used by the ecommerce service.

Reply:
Should we wrap this in `try/catch` to handle the possibility of an `IntegrityError` being raised?

Reply:
Maybe not, since that should only happen when a refund is requested?